Trappist/src/Promact.Trappist.Test/TestConduct/TestConductRepositoryTest.cs

```diff
@@ -201,11 +199,98 @@ public TestConductRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
             var attendeeId = await _trappistDbContext.TestAttendees.Where(x => x.Email.Equals(testAttendee.Email)).Select(x => x.Id).FirstOrDefaultAsync();
             //JSON string
-            var answer = "{ \"questionId\":1,\"optionChoice\":[1],\"code\":null,\"questionStatus\":1}";
+            var answer = "[{ \"questionId\":1,\"optionChoice\":[1],\"code\":null,\"questionStatus\":1}]";
             await _testConductRepository.AddAnswerAsync(attendeeId, answer);
             var attendeeAnswer = await _testConductRepository.GetAnswerAsync(attendeeId);
 
-            Assert.True(attendeeAnswer.QuestionId == 1 && attendeeAnswer.OptionChoice.Count() == 1 && attendeeAnswer.QuestionStatus == QuestionStatus.review);
+            Assert.True(attendeeAnswer.Count == 1);
+        }
+
+        /// <summary>
+        /// Test to get Test Attendee
+        /// </summary>
+        [Fact]
+        public async Task GetTestAttendeeByIdAsyncTest()
+        {
+            var testAttendee = InitializeTestAttendeeParameters();
+            await CreateTestAsync();
+            await _testConductRepository.RegisterTestAttendeesAsync(testAttendee, _stringConstants.MagicString);
+            var attendeeId = await _trappistDbContext.TestAttendees.OrderBy(x => x.Email).Where(x => x.Email.Equals(testAttendee.Email)).Select(x => x.Id).FirstOrDefaultAsync();
+            var attendee = await _testConductRepository.GetTestAttendeeByIdAsync(attendeeId);
+
+            Assert.NotNull(attendee);
+        }
+
+        /// <summary>
+        /// Test to check if Test Attendee exist
+        /// </summary>
+        [Fact]
+        public async Task IsTestAttendeeExistByIdAsyncTest()
+        {
+            var testAttendee = InitializeTestAttendeeParameters();
+            await CreateTestAsync();
+            await _testConductRepository.RegisterTestAttendeesAsync(testAttendee, _stringConstants.MagicString);
+            var attendeeId = await _trappistDbContext.TestAttendees.OrderBy(x => x.Email).Where(x => x.Email.Equals(testAttendee.Email)).Select(x => x.Id).FirstOrDefaultAsync();
+
+            Assert.True(await _testConductRepository.IsTestAttendeeExistByIdAsync(attendeeId));
+        }
+
+        /// <summary>
+        /// Test to set elapsed time 
+        /// </summary>
+        [Fact]
+        public async Task SetElapsedTimeAsyncTest()
+        {
+            var testAttendee = InitializeTestAttendeeParameters();
+            await CreateTestAsync();
+            await _testConductRepository.RegisterTestAttendeesAsync(testAttendee, _stringConstants.MagicString);
+            var attendeeId = await _trappistDbContext.TestAttendees.OrderBy(x => x.Email).Where(x => x.Email.Equals(testAttendee.Email)).Select(x => x.Id).FirstOrDefaultAsync();
+
+            //New entry to AttendeeAnswer table will be made since no answer is saved
+            await _testConductRepository.SetElapsedTimeAsync(attendeeId);
```

Comment:
Yes. For every 3 min a request is made from the client to server to log time elapsed. If the attendee attempts no question for the first 3 minutes the function `SetElapsedTimeAsync()` will add a new entry to the `AttendeeAnswer` table and log the time. The second call will directly log the elapsed time. This will help in code coverage.